src/__tests__/courseIntegration.test.ts

```diff
@@ -0,0 +1,102 @@
+import { gql } from 'apollo-server';
+
+import CourseAPI from '../datasources/course';
+import { EventDetails, EventListResponse } from '../types/types';
+import { getApolloTestServer } from '../utils/testUtils';
+
+it('resolves eventList correctly', async () => {
+  const GET_COURSES = gql`
+    {
+      courseList {
+        data {
+          id
+          name {
+            fi
+          }
+        }
+        meta {
+          count
+        }
+      }
+    }
+  `;
+
+  const courseAPI = new CourseAPI();
+  const eventId = 'eventId';
+  const eventName = 'tapahtuma';
+
+  courseAPI.get = jest.fn().mockResolvedValue({
+    data: [{ id: eventId, name: { fi: eventName } }],
+    meta: { count: 1 },
+  } as EventListResponse);
+
+  const { query } = getApolloTestServer({ dataSources: () => ({ courseAPI }) });
+
+  const res = await query({ query: GET_COURSES });
+
+  expect(res.data.courseList).toEqual({
+    data: [{ id: eventId, name: { fi: eventName } }],
+    meta: {
+      count: 1,
+    },
+  });
+});
+
+it('resolves eventDetails correctly', async () => {
+  const COURSE_DETAILS = gql`
+    {
+      courseDetails {
+        id
+        publisher
+      }
+    }
+  `;
+
+  const courseAPI = new CourseAPI();
+  const eventId = 'eventId';
+  const publisherId = 'publisherId';
```

Comment:
Could have global variables to store these values. They are same for all the tests

Reply:
I will add some mock data utils at some point